tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailboxRepository.scala

```diff
@@ -198,19 +220,26 @@ class TeamMailboxRepositoryImpl @Inject()(mailboxManager: MailboxManager,
       .toSeq)
   }
 
-  override def listMembers(teamMailbox: TeamMailbox): Publisher[Username] = {
+  override def listMembers(teamMailbox: TeamMailbox): Publisher[TeamMailboxMember] = {
     val session: MailboxSession = createSession(teamMailbox)
     SMono.fromPublisher(exists(teamMailbox))
       .filter(b => b)
       .switchIfEmpty(SMono.error(TeamMailboxNotFoundException(teamMailbox)))
       .flatMap(_ => SMono(mailboxManager.listRightsReactive(teamMailbox.mailboxPath, session)))
       .flatMapIterable(mailboxACL => mailboxACL.getEntries.asScala)
-      .map(entryKeyAndRights => entryKeyAndRights._1)
-      .filter(entryKey => NameType.user.equals(entryKey.getNameType))
-      .map(entryKey => Username.of(entryKey.getName))
-      .distinct()
+      .filter(entryKeyAndRights => NameType.user.equals(entryKeyAndRights._1.getNameType))
+      .map(entryKeyAndRights => Username.of(entryKeyAndRights._1.getName) -> entryKeyAndRights._2)
+      .distinct(usernameAndRights => usernameAndRights._1)
+      .map(usernameAndRights => TeamMailboxMember(username = usernameAndRights._1, role = getTeamMemberRole(usernameAndRights._2)))
   }
 
+  private def getTeamMemberRole(rights: MailboxACL.Rfc4314Rights): TeamMemberRole =
```

Comment:
Please handle the case where the rights is fully empty or do not contain the required rights for team mailbox (eg read right)

In which case we likely should filter the value out?